src/main/java/org/openstreetmap/atlas/geography/atlas/sqlite/SQLiteDictionaryWriter.java

```diff
@@ -0,0 +1,52 @@
+package org.openstreetmap.atlas.geography.atlas.sqlite;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+import org.openstreetmap.atlas.geography.atlas.packed.PackedAtlas;
+import org.openstreetmap.atlas.utilities.compression.IntegerDictionary;
+
+/**
+ * Writes the PackedAtlas string dictionary to a table in SQLite.
+ *
+ * @author hallahan
+ */
+public class SQLiteDictionaryWriter
+{
+    private static final String INSERT_SQL = "INSERT INTO dictionary VALUES (?, ?)";
+
+    private final Connection connection;
+    private final PreparedStatement insertStatement;
+    private final PackedAtlasFieldDelegator delegator;
+
+    public SQLiteDictionaryWriter(final Connection connection,
+            final PackedAtlasFieldDelegator delegator) throws SQLException
+    {
+        this.connection = connection;
+        this.insertStatement = this.connection.prepareStatement(INSERT_SQL);
+        this.delegator = delegator;
+    }
+
+    public void write(final PackedAtlas atlas) throws SQLException
+    {
+        final IntegerDictionary<String> dictionary = this.delegator.dictionaryHandle();
+        final int size = dictionary.size();
+
+        // This is very important. When you're doing bulk inserts, you need to turn this off.
+        // Performance is very bad without doing this!
+        this.connection.setAutoCommit(false);
+
+        for (int key = 0; key < size; ++key)
+        {
+            int column = 0;
+            final String value = dictionary.word(key);
+            this.insertStatement.setInt(++column, key);
+            this.insertStatement.setString(++column, value);
+            this.insertStatement.execute();
+        }
+
+        this.connection.commit();
```

Comment:
Are we leaking connections? This is never closed anywhere. Possibly implement `AutoCloseable` here.

Reply:
To add to this comment, I would be very wary of using connections the way you are here by including it as a class variable. I am not sure if there is a need for a connection pool, however you should have a class (like a connection pool) to handle the connections and dole them out to functions as requested. Otherwise you are holding onto that connection the whole time this class is in memory.

Reply:
For something like SQLite it is a little different because it is in-memory. However it is generally considered best practice to retrieve from a connection pool and then close or return connections as soon as you are done with it. For a shared database the benefits of this kind of practice are very obvious. Connection leaks are much more dangerous and can cause your database to run out of open connections making not only your local application start to fail, but also other applications connecting to the database. A connection pool allows you to limit the connections your application makes and shares an open connection among threads within you application. This gives you a bit of performance so that the application doesn't have to continue opening and closing connections to the database. All these things are not as problematic with something like an in-memory database like SQLite. However if your program grows or even if you start using multiple threads within your application, the same principles apply. 

To answer your "Dumb Question" (and it is in quotes because it really is not a dumb question at all, once the JVM exist all resources associated with the JVM will "close". In this case the socket resource to the database will automatically be shut down. However the connection will not be closed properly. A database can react unexpectedly in that scenario and usually the session in the database will be shutdown in a couple of seconds after the socket closes but there may be cases where the session is kept alive for longer than you would ordinarily want. One such case would be if a SQL query is running at the time the JVM exits, it could cause the session to remain alive for a lot longer than expected, especially if you get into a hairy deadlock kind of situation.

Anyway I understand that in this specific instance connection handling won't have as much impact as it would if you were connecting to a shared database resource, but I would still strongly recommend following best practices so that you won't find yourself in a difficult spot if the code outgrows the initial use cases.